app/assets/javascripts/3dbio_viewer/src/webapp/components/protvista/Tooltip.tsx

```diff
@@ -1,38 +1,84 @@
+import { Fragment2, Interval, getConflict } from "../../../domain/entities/Fragment2";
 import { Pdb } from "../../../domain/entities/Pdb";
 import { Subtrack } from "../../../domain/entities/Track";
 import { renderJoin } from "../../utils/react";
 import { Link } from "../Link";
 import { Protein } from "../../../domain/entities/Protein";
+import { trackDefinitions } from "../../../domain/definitions/tracks";
 import { getSource, Source as SourceEntity } from "../../../domain/entities/Source";
 import i18n from "../../utils/i18n";
-import styled from "styled-components";
 
 interface TooltipProps {
     pdb: Pdb;
     subtrack: Subtrack;
     fragment: FragmentP;
+    alignment: Interval[];
     sources: SourceEntity[];
 }
 
 type FragmentP = Fragment | Fragment2;
 
 export const Tooltip: React.FC<TooltipProps> = React.memo(props => {
     //cannot use sources from AppContext as context is not initalized
-    const { pdb, subtrack, fragment, sources } = props;
+    const { pdb, subtrack, fragment, alignment, sources } = props;
     const { description, alignmentScore } = fragment;
 
     //no react.memo as is finally rendered to string
     const nmrSource = getSource(sources, "NMR");
-    const score = alignmentScore ? alignmentScore + " %" : undefined;
+    const score = alignmentScore ? alignmentScore + " %" : undefined; // aligmentScore is never being set on code
+    const isStructureCoverage = subtrack.accession === trackDefinitions.structureCoverage.id;
     const isNMR = subtrack.accession === "nmr";
 
+    const isCovered =
+        isStructureCoverage ||
+        alignment.some(
+            interval => fragment.start >= interval.start && fragment.end <= interval.end
+        );
+
+    const isPartiallyCovered = alignment.some(
+        interval => fragment.start <= interval.end && fragment.end >= interval.start
+    );
+
+    const isNotCovered = alignment.every(
+        interval => fragment.start > interval.end || fragment.end < interval.start
+    );
+
+    const coverage = isNotCovered
+        ? {
+            value: i18n.t("Not in Structure Coverage"),
+            className: "error",
+            title: i18n.t(
+                "This annotation is part of the full UniProt sequence but lies outside the region captured in the 3D structure (PDB) for this specific chain. It may correspond to a flexible, disordered, or missing part of the protein that was not resolved during structure determination."
+            ),
+        }
+        : isPartiallyCovered && !isCovered
+            ? {
+                value: i18n.t("Partially in Structure Coverage"),
+                className: "warning",
+                title: i18n.t(
+                    "This annotation is partially covered by the 3D structure (PDB) for this specific chain. The structure may capture part of this region, but additional functional or structural details extend beyond the resolved area."
+                ),
+            }
+            : undefined;
+
```

Comment:
Not a big fan of ternaries with more than one operation, but it looks quite readable here, so let's keep it.

I know the problem, if you decide to move it to a if/else if/else chain, you are forced to use functions or ugly IIFEs. That why in other languages (specially functional ones), ifs are expressions, not statements.